indico/modules/events/contributions/client/js/index.jsx

```diff
@@ -8,13 +8,40 @@
 /* global showUndoWarning:false, setupListGenerator:false, setupSearchBox:false */
 /* global reloadManagementAttachmentInfoColumn:false */
 
+import fileTypesURL from 'indico-url:event_editing.api_file_types';
+
 import React from 'react';
 import ReactDOM from 'react-dom';
 
 import 'indico/modules/events/util/types_dialog';
-import {PublicationSwitch} from 'indico/react/components';
+import {EditableSubmissionButton, PublicationSwitch} from 'indico/react/components';
+
+import {camelizeKeys} from 'indico/utils/case';
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
 
 (function(global) {
+  global.setupEditableSubmissionButton = async function setupEditableSubmissionButton() {
+    const editableSubmissionButton = document.querySelector('#editable-submission-button');
+    const {eventId, apiEditableSubmitUrl, type, uploadUrl} = editableSubmissionButton.dataset;
+
+    let response;
+    try {
+      response = await indicoAxios.get(fileTypesURL({confId: eventId}));
+    } catch (e) {
+      handleAxiosError(e);
```

Comment:
return in this case to avoid breaking the code below which would get an undefined response